lib/dialects/sqlite3/schema/ddl.js

```diff
@@ -344,6 +344,40 @@ class SQLite3_DDL {
     );
   }
 
+  setNullable(column, isNullable) {
+    return this.client.transaction(
+      async (trx) => {
+        this.trx = trx;
+
+        const { createTable, createIndices } = await this.getTableSql();
+
+        const parsedTable = parseCreateTable(createTable);
+        const parsedColumn = parsedTable.columns.find((c) =>
+          isEqualId(column, c.name)
+        );
+
+        if (!parsedColumn) {
+          throw new Error(
+            `.setNullable: Column ${column} does not exist in table ${this.tableName()}.`
+          );
+        }
+
+        parsedColumn.constraints.notnull = isNullable
+          ? null
+          : { name: null, conflict: null };
+
+        parsedColumn.constraints.null = isNullable
+          ? parsedColumn.constraints.null
+          : null;
+
+        const newTable = compileCreateTable(parsedTable, this.wrap);
+
+        return this.alter(newTable, createIndices);
```

Comment:
It is preferable to avoid directly executing SQL inside `setNullable` and return an array of generated SQL statements instead.

Reply:
I see this as an extension of your point on `statementsProducer` so I'll mark this one as resolved.